packages/opentelemetry-core/src/common/id-util.ts

```diff
@@ -0,0 +1,35 @@
+/**
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as crypto from 'crypto';
```

Comment:
Did the SIG decide that `opentelemetry-core` is Node specific? That's fine I think and matches the OpenCensus approach but wanted to at least make sure we were being clear on that.

I recognize that the browser based tracer is a secondary use case and I'm open to basically re-implementing the `core` package in a web-centric and web-optimized way, just by sharing the TypeScript types. But want to make sure we are making an intentional decision on that.

Reply:
I actually totally missed that but it's true that `crypto` should be extracted to some kind of `platform-node`. IMO this should be the case regardless of a shared core VS multiple cores.

Reply:
> The main alternative seems to be something where the platform-specific parts of the code are abstracted away and keep the core package platform neutral and then maybe have core-node and core-web that would inject the platform-specific implementations of things like random number generation, timestamp collection, etc.

Thanks for the feedback, makes sense to me. I really don't have experience in handling these use-cases. It would be nice and useful if you can lay the groundwork/foundation.

Reply:
Personally I would move `randomBytes` to a `platform-node` package that we can later on have a `platform-browser` with the same API. Then generate a seed from that.

I would then use a the result of `Math.random()` stored in a `Uint32Array` for the IDs. This approach is very fast and safe given the truly random seed and the Xorshift128+ algorithm used by `Math.random().`

Reply:
What about merging it as it is and refactoring the parts out right after?
Just to keep iterating.

Reply:
Definitely. At this point in time I think most comments should not block PRs but instead we can move forward and open issues to not lose track of these conversations.